### PR TITLE
CI: Add mambaforge to ci builds

### PR DESCRIPTION
**Pull request by phofl**

### CODE REVIEW

**Comment by mroeschke** [comment]
Nice! Any sense of how much faster the environment building is?

**Comment by phofl** [comment]
The conda incubator takes around 3 minutes I think. With conda it takes around 4 in a best case scenario. Unfortunately we have a lot of builds where this takes more than 15, while mamba should stay at around 3 for them. It will keep our build times consistent.

Unfortunately we have 2 failing pipelines which do seem relevant, will check what this is about

**Comment by jonashaag** [comment]
Also add here https://github.com/pandas-dev/pandas/blob/master/.github/workflows/sdist.yml#L54 and here https://github.com/pandas-dev/pandas/blob/master/.github/workflows/asv-bot.yml#L44?

**Comment by phofl** [comment]
This is not finished yet. I will have to figure out how to fix the installation issues before adding this somewhere else.

**Comment by jonashaag** [comment]
FYI, on my machine (macOS) those tests run fine when using the script.

**Comment by phofl** [comment]
This is not a test issue per se. Mamba resolves the dependencies differently, which results in different versions. The 3.9 build installs arrow 2.0.1 instead of 6.0.0, causing the tests to fail. I think I found a solution for this, but have to look a bit further

**Comment by jonashaag** [comment]
What I meant to say is that for me Mamba seems to resolve correctly/in a way that doesn't break the tests.

**Comment by phofl** [comment]
With the 3.9 dependency file? Could you post the output of conda info?

**Comment by jonashaag** [comment]
@phofl see #44979, seems to work ok

**Comment by phofl** [comment]
Yeah, would want to avoid the pin, but we'll see. Will finish this tomorrow after the setuptools pin is merged

**Comment by phofl** [comment]
This is ready for review.

We have a few builds where creating the environment regularly takes more than one hour. Mamba does speed up the process a bit in general, but more important: it does not take forever in some instances.

The 3.9 envs are solved a bit weird. Hence I added the pyarrow pin, which avoids this

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -41,6 +41,8 @@ jobs:
 
     - uses: conda-incubator/setup-miniconda@v2
       with:
+        mamba-version: "*"
+        channels: conda-forge,defaults
```
Comment (lithomas1) [inline .github/workflows/ci.yml:45]: Can we avoid adding defaults as a channel here and elsewhere, if possible? Having multiple channels tends to slow things down a bit.

Comment (phofl) [reply .github/workflows/ci.yml]: Yep, works.

**Comment by jreback** [comment]
do we still have pyarrow 1.* testing?

**Comment by phofl** [comment]
Yep, for example in Database / Linux_py38_IO (ci/deps/actions-38-db-min.yaml)

**Review by lithomas1 — APPROVED** [review]

**Comment by jreback** [comment]
thanks @phofl

**Pull request merged**